### ISSUE TITLE
question on loss function of VAE

### ISSUE THREAD

**Issue by Ishihara-Masabumi** [masato-ka/airc-rl-agent#44]
Loss function of VAE is as follows:

```
    def loss_fn(self, images, reconst, mean, logvar):
        KL = -0.5 * torch.sum((1 + logvar - mean.pow(2) - logvar.exp()), dim=0)
        KL = torch.mean(KL)
        reconstruction = F.binary_cross_entropy(reconst.view(-1,38400), images.view(-1, 38400), reduction='sum') #size_average=False)
        return reconstruction + 5.0 * KL
```

I can see the part of the loss, "reconstruction".
But, what is the meaning of "5.0 * KL"?

**Comment by masato-ka** [comment]
KL is the KL divergence term, which is responsible for constraining the latent space of the VAE to a regular expression.
5.0 is a beta parameter. It is a coefficient for disentanglement the latent space. disentanglement is to make each element of the latent space have an independent meaning. For example, the value of one element can be constrained to affect only the background.

detail is see in https://openreview.net/pdf?id=Sy2fzU9gl

**Comment by masato-ka** [comment]
Could I close this question ?

**Comment by Ishihara-Masabumi** [comment]
I don't fully understand it, but I will read and understand the paper you recommended.
Therefore, you can close this issue.
